core/src/providers/openai.rs

```diff
@@ -1200,12 +1200,17 @@ impl Embedder for OpenAIEmbedder {
             ));
         }
 
-        // Give priority to `CORE_DATA_SOURCES_OPENAI_API_KEY` env variable
-        match std::env::var("CORE_DATA_SOURCES_OPENAI_API_KEY") {
-            Ok(key) => {
-                self.api_key = Some(key);
-            }
-            Err(_) => match credentials.get("OPENAI_API_KEY") {
+        // Give priority to `CORE_DATA_SOURCES_OPENAI_API_KEY` env variable.
+        let core_key = std::env::var("CORE_DATA_SOURCES_OPENAI_API_KEY").ok();
+        let use_eu_endpoint: bool;
+
+        if let Some(key) = core_key {
+            self.api_key = Some(key);
+            // Never use EU endpoint when using CORE_DATA_SOURCES_OPENAI_API_KEY.
+            use_eu_endpoint = false;
```

Comment:
Isn't this getting overridden after?

And isn't the env variable always set in production environment?